sample-package-core/src/main/java/org/nuxeo/sample/core/CustomBulkAction.java

```diff
@@ -0,0 +1,39 @@
+package org.nuxeo.sample.core;
+
+import static org.nuxeo.ecm.core.bulk.BulkServiceImpl.STATUS_STREAM;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.INPUT_1;
+import static org.nuxeo.lib.stream.computation.AbstractComputation.OUTPUT_1;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.bulk.action.computation.AbstractBulkComputation;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class CustomBulkAction implements StreamProcessorTopology {
+  public static final String ACTION_NAME = "custombulkaction";
+
+  @Override
+  public Topology getTopology(Map<String, String> map) {
+    return Topology.builder()
+    .addComputation(CustomBulkActionComputation::new,
+    Arrays.asList(INPUT_1 + ":" + ACTION_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+    .build();
+    }
+
+  public static class CustomBulkActionComputation extends AbstractBulkComputation {
+
+    public CustomBulkActionComputation() {
+      super("bulk/" + ACTION_NAME);
+    }
+
+    @Override
+    protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+      // TODO: implement code to execute against document ids
```

Comment:
maybe gives more hints:
you will receive a list of document ids that can be loaded using 
// DocumentModelList docs = loadDocuments(session, ids);
the number of document ids is limited to batchSize and a transaction is managed around the `compute` call

Reply:
and add emphasis on the contract: *you should catch any expected errors*:
- a document could have been deleted, 
- you might not have the write permission
- if you call a third part service always set timeout and catch any possible error
you can report errors to the bulk command status, using:
```delta.inError(numberOfErrorInTheBatch, someErrorMessage)```

Note that in case of failure (uncatched exception) the retry mechanism at the computation level will operate
but this can create duplicate processing furthermore if the failure is systematic the bulk command will never be in completed status.

Reply:
Comments added to give more hints to developers, thank you!